api_tests/institutions/views/test_institution_relationship_nodes.py

```diff
@@ -323,3 +330,265 @@ def test_delete_user_is_affiliated_with_inst_and_mixed_permissions_on_nodes(self
+        registration.save()
+        return registration
+
+    @pytest.fixture()
+    def registration_public(self, admin, institution):
+        registration = RegistrationFactory(creator=admin, is_public=True)
+        registration.affiliated_institutions.add(institution)
+        registration.save()
+        return registration
+
+    @pytest.fixture()
+    def url_institution_registrations(self, institution):
+        return '/{}institutions/{}/relationships/registrations/'.format(API_BASE, institution._id)
+
+
+    def test_auth_get_registrations(self, app, admin, registration_no_owner, registration_no_affiliation, registration_pending, registration_public, url_institution_registrations):
+
+        #test getting registrations without auth (for complete registrations)
+        res = app.get(url_institution_registrations)
+        assert res.status_code == 200
+        registration_ids = [reg['id'] for reg in res.json['data']]
+        assert registration_no_affiliation._id not in registration_ids
+        assert registration_pending._id not in registration_ids
+        assert registration_no_owner._id not in registration_ids
+        assert registration_public._id in registration_ids
+
+        #Withdraw a registration, make sure it still shows up
+        WithdrawnRegistrationFactory(registration=registration_public, user=admin)
+
+        #test getting registrations with auth (for embargoed and pending)
+        res = app.get(url_institution_registrations, auth=admin.auth)
+        assert res.status_code == 200
+        registration_ids = [reg['id'] for reg in res.json['data']]
+        assert registration_no_affiliation._id not in registration_ids
+        assert registration_pending._id in registration_ids
+        assert registration_public._id in registration_ids
+        assert registration_no_owner._id not in registration_ids
+
+    def test_add_on_nonexistent_registration(self, app, admin, registration_no_affiliation, url_institution_registrations):
+        #test registration does not exist
+        res = app.post_json_api(
+            url_institution_registrations,
+            make_registration_payload('notIdatAll'),
+            expect_errors=True,
+            auth=admin.auth
+        )
+
+        assert res.status_code == 404
+
+        res = app.post_json_api(
+            url_institution_registrations,
+            {'data': [{'type': 'nodes', 'id': registration_no_affiliation._id}]},
+            expect_errors=True,
+            auth=admin.auth
+        )
+
+        assert res.status_code == 409
+
+    def test_add_no_permissions(self, app, registration_no_affiliation, url_institution_registrations, institution):
+        res = app.post_json_api(
+            url_institution_registrations,
+            make_registration_payload(registration_no_affiliation._id),
+            expect_errors=True,
+            auth = AuthUserFactory().auth
+        )
+
+        assert res.status_code == 403
```

Comment:
Two things: 

1. All the 400-series checks could be combined into a single test to save on the amount of time it takes to set up each test;
2. I think you're missing a no-authentication 401 test.